call/constants.go

```diff
@@ -1,5 +1,54 @@
 package call
 
+type Chain string
+
+type ChainConfig struct {
+	MultiCallAddress string
+	Url              string
+}
+
 const (
-	BinanceChain = "0x41263cBA59EB80dC200F3E2544eda4ed6A90E76C"
-)
\ No newline at end of file
+	Arbitrum  Chain = "Arbitrum"
+	Aurora          = "aurora"
+	Avalanche       = "avalanche"
+	Bsc             = "bsc"
+	Ethereum        = "ethereum"
+	Fantom          = "fantom"
+	Moonbeam        = "moonbeam"
+	Moonriver       = "moonriver"
+)
+
+var Chains = map[Chain]ChainConfig{
+	Arbitrum: {
+		MultiCallAddress: "0x7a7443f8c577d537f1d8cd4a629d40a3148dd7ee",
+		Url:              "https://arb1.arbitrum.io/rpc",
+	},
+	Aurora: {
+		MultiCallAddress: "0x88b373B83166E72FD55648Ce114712633f1782E2",
+		Url:              "https://mainnet.aurora.dev",
+	},
+	Avalanche: {
+		MultiCallAddress: "0xa00FB557AA68d2e98A830642DBbFA534E8512E5f",
+		Url:              "https://api.avax.network/ext/bc/C/rpc",
+	},
+	Bsc: {
+		MultiCallAddress: "0x41263cBA59EB80dC200F3E2544eda4ed6A90E76C",
+		Url:              "https://bsc-dataseed1.ninicoin.io",
+	},
+	Ethereum: {
+		MultiCallAddress: "0xeefba1e63905ef1d7acba5a8513c70307c1ce441",
+		Url:              "https://mainnet.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161",
+	},
+	Fantom: {
+		MultiCallAddress: "0x7F4e475462A0fA0F1e2C69d50866D54505F99D72",
+		Url:              "https://rpcapi.fantom.network",
+	},
+	Moonbeam: {
+		MultiCallAddress: "0x6477204E12A7236b9619385ea453F370aD897bb2",
+		Url:              "https://moonbeam.public.blastapi.io",
+	},
+	Moonriver: {
+		MultiCallAddress: "0xaef00a0cf402d9dedd54092d9ca179be6f9e5ce3",
+		Url:              "https://moonriver.public.blastapi.io",
+	},
+}
```

Comment:
```suggestion
type ChainConfigList struct {
	Arbitrum  ChainConfig
	Aurora    ChainConfig
	Avalanche ChainConfig
	Bsc       ChainConfig
	Ethereum  ChainConfig
	Fantom    ChainConfig
	Moonbeam  ChainConfig
	Moonriver ChainConfig
}

var Chains = ChainConfigList{
	Arbitrum: ChainConfig{
		MultiCallAddress: "0x7a7443f8c577d537f1d8cd4a629d40a3148dd7ee",
		Url:              "https://arb1.arbitrum.io/rpc",
	},
	Aurora: ChainConfig{
		MultiCallAddress: "0x88b373B83166E72FD55648Ce114712633f1782E2",
		Url:              "https://mainnet.aurora.dev",
	},
	Avalanche: ChainConfig{
		MultiCallAddress: "0xa00FB557AA68d2e98A830642DBbFA534E8512E5f",
		Url:              "https://api.avax.network/ext/bc/C/rpc",
	},
	Bsc: ChainConfig{
		MultiCallAddress: "0x41263cBA59EB80dC200F3E2544eda4ed6A90E76C",
		Url:              "https://bsc-dataseed1.ninicoin.io",
	},
	Ethereum: ChainConfig{
		MultiCallAddress: "0xeefba1e63905ef1d7acba5a8513c70307c1ce441",
		Url:              "https://mainnet.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161",
	},
	Fantom: ChainConfig{
		MultiCallAddress: "0x7F4e475462A0fA0F1e2C69d50866D54505F99D72",
		Url:              "https://rpcapi.fantom.network",
	},
	Moonbeam: ChainConfig{
		MultiCallAddress: "0x6477204E12A7236b9619385ea453F370aD897bb2",
		Url:              "https://moonbeam.public.blastapi.io",
	},
	Moonriver: ChainConfig{
		MultiCallAddress: "0xaef00a0cf402d9dedd54092d9ca179be6f9e5ce3",
		Url:              "https://moonriver.public.blastapi.io",
	},
}
```

I think when we call `NewContractBuilder().ChainConfig(Chains.Ethereum)` is easier to use than `NewContractBuilder().ChainConfig(Chains[call.Ethereum])`

Reply:
Actually, you can simply use the following: `NewContractBuilder().DefaultConfig("bsc")` or `NewContractBuilder().DefaultConfig("ethereum")`. So I don't think we need to change here.

Reply:
Ok, I thought some one may have a typo. But that not bad when we also have this [line of code](https://github.com/depocket/multicall-go/pull/2/files#diff-d8afb10d455139795d89e3774c9605942beccb92f7efca5a77de754a8c484c00R58) as a validator.

---

An other question, `DefaultConfig("bsc")`'s mean doesn't present for the default in this context of the code, It present for the supported chains, right?
If so, how do you think about the `WithChain('bsc')` or `OnChain('bsc')`

Reply:
Follow on https://github.com/depocket/multicall-go/pull/2#discussion_r896445070